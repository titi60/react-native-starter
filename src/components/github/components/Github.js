import React, { Component } from 'react';
import { Image } from 'react-native';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';

import getTranslations from 'utils/locales';

import Box from 'shared/Box';
import Text from 'shared/Text';
import Button from 'shared/Button';

const query = gql`
  {
    viewer {
      login
      avatarUrl
    }
  }
`;

class Github extends Component {
  goBack = () => {
    const { navigation } = this.props;

    navigation.goBack(null);
  };

  render() {
    return (
      <Box flex={1} justifyContent="center" alignItems="center">
        <Query query={query}>
          {({ loading, error, data }) => {
            if (loading) return <Text>Loading...</Text>;
            if (error) return <Text>Some error happened :</Text>;

            const {
              viewer: { login, avatarUrl },
            } = data;

            return (
              <>
                <Text fontSize={4} mb={3}>
                  {getTranslations('github', 'page_title')}
                </Text>

                <Image
                  source={{ uri: avatarUrl }}
                  style={{ width: 100, height: 100, borderRadius: 50 }}
                />

                <Text fontSize={3} mt={3}>
                  {`${getTranslations(
                    'github',
                    'profil_name',
                    {
                      name: login,
                    },
                    'name'
                  )} ${login}`}
                </Text>

                <Button onPress={this.goBack}>
                  <Text mt={3}>
                    {getTranslations('github', 'navigation_back')}
                  </Text>
                </Button>
              </>
            );
          }}
        </Query>
      </Box>
    );
  }
}

export default Github;
