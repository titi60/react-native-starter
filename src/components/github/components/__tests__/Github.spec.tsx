import React from 'react';
import { shallow } from 'enzyme';

import Github from '../Github';

describe('Github page component', () => {
  // given
  const props = { navigation: { goBack: jest.fn() } as any };

  it('should render correctly', () => {
    // When
    const wrapper = shallow(<Github {...props} />);

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
