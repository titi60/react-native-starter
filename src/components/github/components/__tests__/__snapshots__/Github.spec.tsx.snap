// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Github page component should render correctly 1`] = `
<Styled(View)
  alignItems="center"
  flex={1}
  justifyContent="center"
>
  <Query
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": undefined,
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "viewer",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "login",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "avatarUrl",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 56,
          "start": 0,
        },
      }
    }
  >
    <Component />
  </Query>
</Styled(View)>
`;
